File: backend/app/schemas/stock.py
from datetime import date
from pydantic import BaseModel


class PriceBar(BaseModel):
    date: date
    open: float | None
    high: float | None
    low: float | None
    close: float | None
    volume: int | None


class PriceHistoryResponse(BaseModel):
    ticker: str
    bars: list[PriceBar]


class TickerInfoResponse(BaseModel):
    ticker: str
    company_name: str | None
    current_price: float | None
    sector: str | None
    market_cap: float | None
    currency: str | None

File: frontend/src/types/stock.ts
export interface WatchlistTicker {
  id: number
  ticker: string
  company_name: string | null
  sector: string | null
  market_cap: number | null
  currency: string | null
  added_at: string
  is_active: boolean
  notes: string | null
}

export interface PriceBar {
  date: string
  open: number | null
  high: number | null
  low: number | null
  close: number | null
  volume: number | null
}

export interface PriceHistoryResponse {
  ticker: string
  bars: PriceBar[]
}

export interface TickerInfo {
  ticker: string
  company_name: string | null
  current_price: number | null
  sector: string | null
  market_cap: number | null
  currency: string | null
}
